concourse/client/util.py

```diff
@@ -122,3 +152,159 @@ def jobs_not_triggered(
             for pr_resource_version in pr_resource_versions:
                 if not wait_for_job_to_be_triggered(job, pr_resource_version, concourse_api):
                     yield (job, pr_resource, pr_resource_version)
+
+
+def get_resource_lock_and_trigger_build(
+    job: Job,
+    resource: PipelineConfigResource,
+    resource_version: ResourceVersion,
+    concourse_api,
+    max_retries: int=3,
+):
+    retries = max_retries
+    while retries >= 0:
+        try:
+            with pin_resource(
+                resource=resource,
+                resource_version=resource_version,
+                concourse_api=concourse_api,
+            ):
+                concourse_api.trigger_build(
+                    pipeline_name=resource.pipeline_name(),
+                    job_name=job.pipeline_name,
+                )
+
+                job_retry = 10
```

Comment:
nit: the plural form is written `job_retries`

Reply:
inline not possible we use it 2 times